### PR TITLE
Fix documentation around JSON and content-type (#2598)

### PR DESCRIPTION
**Pull request by socketpair**

### CODE REVIEW

File: docs/client_advanced.rst [hunk 1]
```diff
@@ -26,27 +26,35 @@ Custom Request Headers
 If you need to add HTTP headers to a request, pass them in a
 :class:`dict` to the *headers* parameter.
 
-For example, if you want to specify the content-type for the previous
-example::
```
Comment (socketpair) [inline docs/client_advanced.rst:30]: I did not find `previous example`

File: docs/client_advanced.rst [hunk 2]
```diff
@@ -26,27 +26,35 @@ Custom Request Headers
...
+For example, if you want to specify the content-type directly::
 
-    import json
-    url = 'https://api.github.com/some/endpoint'
-    payload = {'some': 'data'}
-    headers = {'content-type': 'application/json'}
```
Comment (socketpair) [inline docs/client_advanced.rst:35]: Wrong way to specify content-type for this (JSON) example. It's obsolete.

File: docs/client_advanced.rst [hunk 3]
```diff
@@ -26,27 +26,35 @@ Custom Request Headers
...
 
+Typical usecase is sending JSON body. You can specify content type
+directly as shown above, but it is more convenient to use special keyword
+``json``::
+
+    await session.post(url, json={'example': 'text'})
+
+The same for *text/plain*::
+
+    await session.post(url, text='Привет, Мир!')
```
Comment (socketpair) [inline docs/client_advanced.rst:57]: Shows that national alphabets work.

File: docs/client_quickstart.rst [hunk 4]
```diff
@@ -251,18 +251,22 @@ dictionary of data will automatically be form-encoded when the request is made::
     }
 
 If you want to send data that is not form-encoded you can do it by
-passing a :class:`str` instead of a :class:`dict`. This data will be
```
Comment (socketpair) [inline docs/client_quickstart.rst:254]: According to documentation it is not allowed to pass `str` here.

File: docs/client_advanced.rst (hunk 1 above)
Comment (asvetlov) [reply docs/client_advanced.rst:30]: I think it was moved to quickstart section.

File: docs/client_advanced.rst (hunk 2 above)
Comment (asvetlov) [reply docs/client_advanced.rst:35]: Agree.

**Comment by asvetlov** [comment]
@socketpair please merge when green

File: docs/client_advanced.rst [hunk 5]
```diff
@@ -26,27 +26,35 @@ Custom Request Headers
...
-                       data=json.dumps(payload),
+                       data=payload,
                        headers=headers)
 
 You also can set default headers for all session requests::
 
-    async with aiohttp.ClientSession(
-        headers={"Authorization": "Basic bG9naW46cGFzcw=="}) as session:
+    headers={"Authorization": "Basic bG9naW46cGFzcw=="}
+    async with aiohttp.ClientSession(headers) as session:
```
Comment (socketpair) [inline docs/client_advanced.rst:43]: BUG it should be `headers=headers` :(

Comment (asvetlov) [reply docs/client_advanced.rst:43]: Fixed by 1cb4fa27

**Pull request merged**